### PR TITLE
Fail test if the expected and actual traces differ (#777)

### PR DESCRIPTION
**Pull request by burmanm**
Use pretty.Diff(expected, actual) instead of reflect.DeepEqual(expected, actual) to check for equality between given and received trace. Also, if there is a difference, set the test to fail instead of only logging the failed trace.

And prettify the storage-integration-test output

### CODE REVIEW

File: plugin/storage/integration/domain_trace_compare_test.go [hunk 1]
```diff
@@ -50,13 +50,14 @@ func CompareTraces(t *testing.T, expected *model.Trace, actual *model.Trace) {
 	model.SortTrace(expected)
 	model.SortTrace(actual)
 	checkSize(t, expected, actual)
-	if !assert.EqualValues(t, expected, actual) {
-		for _, err := range pretty.Diff(expected, actual) {
-			t.Log(err)
-		}
+
+	diff := pretty.Diff(expected, actual)
+	if len(diff) > 0 {
```
Comment (yurishkuro) [inline plugin/storage/integration/domain_trace_compare_test.go]: nit: combine into one line: `if diff := pretty.Diff(expected, actual); len(diff) > 0 {`

File: plugin/storage/integration/domain_trace_compare_test.go [hunk 2]
```diff
@@ -50,13 +50,14 @@ func CompareTraces(t *testing.T, expected *model.Trace, actual *model.Trace) {
...
+		t.Logf("Expected and actual differ: %v\n", diff)
```
Comment (yurishkuro) [inline plugin/storage/integration/domain_trace_compare_test.go]: when there are many differences, this doesn't print them in easily readable way. I prefer the loop as before.

**Comment by burmanm** [comment]
Yep, modified based on the review.

**Comment by yurishkuro** [comment]
looks like the builds are failing due to #786

File: plugin/storage/integration/domain_trace_compare_test.go [hunk 3]
```diff
@@ -50,13 +50,15 @@ func CompareTraces(t *testing.T, expected *model.Trace, actual *model.Trace) {
 	model.SortTrace(expected)
 	model.SortTrace(actual)
 	checkSize(t, expected, actual)
-	if !assert.EqualValues(t, expected, actual) {
-		for _, err := range pretty.Diff(expected, actual) {
-			t.Log(err)
+
+	if diff := pretty.Diff(expected, actual); len(diff) > 0 {
+		for _, d := range diff {
+			t.Logf("Expected and actual differ: %v\n", d)
```
Comment (yurishkuro) [inline plugin/storage/integration/domain_trace_compare_test.go]: nit: `\n` is not necessary, but ok

**Comment by yurishkuro** [comment]
thanks @burmanm !

**Pull request merged**